src/components/UserProfile/UserProfile.jsx

```diff
@@ -290,7 +290,10 @@ const UserProfile = props => {
     }
     try {
       await props.updateUserProfile(props.match.params.userId, userProfile);
-      await props.refreshToken(userProfile._id);
+
+      if (userProfile._id == props.auth.user.userid) {
+        await props.refreshToken(userProfile._id);
```

Comment:
In general one of the best practise is to reduce no of API calls. I think there is not need to update the permissions unless the role is changed. We can include one more check to see if role is changed as well to avoid unnecessary API calls, state changes and hence the re-renderings.

Reply:
@RJBornToWin the change that you requested was made, thank you for the hint.